arrows/vxl/convert_image.cxx

```diff
@@ -0,0 +1,508 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "convert_image.h"
+
+#include <arrows/vxl/image_container.h>
+
+#include <vil/vil_convert.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_math.h>
+#include <vil/vil_plane.h>
+
+#include <cstdlib>
+#include <limits>
+#include <type_traits>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+namespace {
+
+// Convert a floating point image to an intergral type by multiplying
+// it by a scaling factor in addition to thresholding it in one operation.
+// Performs rounding.
+template < typename InType, typename OutType >
+void
+scale_image( const vil_image_view< InType >& src,
+             vil_image_view< OutType >& dst,
+             const double& dp_scale )
+{
+  // Resize, cast and copy
+  unsigned ni = src.ni(), nj = src.nj(), np = src.nplanes();
+  dst.set_size( ni, nj, np );
+
+  std::ptrdiff_t sistep = src.istep(), sjstep = src.jstep(),
+                 spstep = src.planestep();
+  std::ptrdiff_t distep = dst.istep(), djstep = dst.jstep(),
+                 dpstep = dst.planestep();
+
+  const OutType max_output_value = std::numeric_limits< OutType >::max();
+
+  const InType max_input_value = static_cast< InType >(
+    static_cast< double >( max_output_value ) / dp_scale );
+
+  const InType scale = static_cast< InType >( dp_scale );
+
+  const InType* splane = src.top_left_ptr();
+  OutType* dplane = dst.top_left_ptr();
+
+  for( unsigned p = 0; p < np; ++p, splane += spstep, dplane += dpstep )
+  {
+    const InType* srow = splane;
+    OutType* drow = dplane;
+    for( unsigned j = 0; j < nj; ++j, srow += sjstep, drow += djstep )
+    {
+      const InType* spixel = srow;
+      OutType* dpixel = drow;
+      for( unsigned i = 0; i < ni; ++i, spixel += sistep, dpixel += distep )
+      {
+        if( *spixel <= max_input_value )
+        {
+          *dpixel = static_cast< OutType >( *spixel * scale + 0.5 );
+        }
+        else
+        {
+          *dpixel = max_output_value;
+        }
+      }
+    }
+  }
+}
+
+template < typename Type >
+void
+combine_channels( const vil_image_view< Type >& src,
```

Comment:
pedantic: please use `T const` instead of `const T`

Reply:
Could this be tool-enforced?

Reply:
I'm guessing based on your [open uncrustify feature request](https://github.com/uncrustify/uncrustify/issues/1952), that it can't at the moment.